### PR TITLE
3988767 persistent postsurvey

### PR DESCRIPTION
**Pull request by elopezanaya**
### Id of the task, bug, story or other reference.
3988767
### Description
_Include a description of the problem to be solved_
When a persistent chat is closed from the customer side, is presenting the postsurvey.

The conversation should be ended and survey presented only when agent ends the conversation.

## Solution Proposed
_Detail what is the solution proposed, include links to design document if required or any other document required to support the solution_

Handle end close workflow for persistent chat when closed by the customer

### Acceptance criteria
_Define what are the conditions to consider the PR has achieved the intended goal_
- close chat for persistent chat should present the survey only when ended by the agent
- when ended by customer should not end the chat or present survey

## Test cases and evidence
_Include what tests cases were considered, any evidence of testing for future references, to identify any corner cases, etc_

- persistent chat , ended by agent should present survey
- persistent chat, ended by customer should not present survey
- close chat with SDK event should act the same as the button
- validate no persistent chat scenarios

- normal chat, insert link (ended by agent and ended by customer)
- 
image

- normal chat, embedded (ended by agent and ended by customer)
 
image

[Recording-20240426_183409.webm](https://github.com/microsoft/omnichannel-chat-widget/assets/981914/aee64d7f-a2cc-4f85-838b-5f155c2fb7da)

- normal chat validated

### Sanity Tests
-   [ ] You have tested all changes in Popout mode
-   [ ] You have tested all changes in cross browsers i.e Edge, Chrome, Firefox, Safari and mobile devices(iOS and Android)
-   [ ] Your changes are included in the [CHANGELOG](../CHANGE_LOG.md)

## A11y
- [ ] You have run the [Automated Accessibility Check](https://accessibilityinsights.io/docs/en/windows/getstarted/automatedchecks) and have no reported issues

__*Please provide justification if any of the validations has been skipped.*__

### CODE REVIEW

**Comment by elopezanaya** [comment]
Retest this please

**Review by matan-hershberg-nuance — APPROVED** [review]

File: chat-widget/src/components/livechatwidget/common/endChat.ts [hunk 1]
```diff
@@ -72,13 +74,23 @@ const prepareEndChat = async (props: ILiveChatWidgetProps, chatSDK: any, state:
                 Description: `${PrepareEndChatDescriptionConstants.ConversationEndedByCustomerWithInvalidPostChat} ${state?.appStates?.conversationEndedBy}.`
             });
         }
-        endChat(props, chatSDK, state, dispatch, setAdapter, setWebChatStyles, adapter, false, true, true);
```
Comment (mithunjohny) [inline chat-widget/src/components/livechatwidget/common/endChat.ts:77]: maybe can make little more concise .
`const { chatConfig, chatSDK } = props;
const { appStates } = executeReducer(state, { type: LiveChatWidgetActionType.GET_IN_MEMORY_STATE, payload: undefined });

const persistentEnabled = isPersistentEnabled(chatConfig);
const endedByCustomer = appStates?.conversationEndedBy === "Customer";

const commonParams = [props, chatSDK, state, dispatch, setAdapter, setWebChatStyles, adapter];

if (persistentEnabled && endedByCustomer) {
    await endChat(...commonParams, true, false, true);
} else {
    await endChat(...commonParams, false, true, true);
}`

**Review by mithunjohny — APPROVED** [review]
